server/src/main/java/org/elasticsearch/index/engine/RecoverySourcePruneMergePolicy.java

```diff
@@ -258,6 +247,48 @@ public StoredFieldsReader clone() {
 
         }
 
+        private static class FilterStoredFieldVisitor extends StoredFieldVisitor {
```

Comment:
Do we need this new private class when we already have one in `org.elasticsearch.search.internal`?